src/tutorial/testing.md

```diff
@@ -82,20 +82,169 @@ Worse, it often deals with user input,
 reads files,
 and writes output.
 
+## Making you code testable
+
+There are two complementary approaches to testing functionality:
+Testing the small units that you build your complete application from,
+and testing the final application "from the outside".
+Let's begin with the first one.
```

Comment:
The two types of testing should probably be named here like 'Testing the small units that you build your complete application from, these are called "unit tests". There is also testing the final application "from the outside" called "black box tests".'